src/main/java/teammates/ui/controller/InstructorCourseInstructorEditSaveAction.java

```diff
@@ -29,10 +28,33 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         gateKeeper.verifyAccessible(instructor, logic.getCourse(courseId),
                                     Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR);
 
-        InstructorAttributes instructorToEdit =
-                extractUpdatedInstructor(courseId, instructorId, instructorName, instructorEmail);
-        updateToEnsureValidityOfInstructorsForTheCourse(courseId, instructorToEdit);
+        String instructorId = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_ID);
+
+        // get instructor without making any edits
+        InstructorAttributes instructorToEdit;
+        if (instructorId == null) {
+            instructorToEdit = logic.getInstructorForEmail(courseId, instructorEmail);
+        } else {
+            instructorToEdit = logic.getInstructorForGoogleId(courseId, instructorId);
+        }
+
+        boolean isDisplayedToStudents = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_IS_DISPLAYED_TO_STUDENT)
+                != null;
+        int numOfInstrDisplayed = getNumberOfInstructorsDisplayedToStudents(
+                courseId, instructorToEdit.isDisplayedToStudents, isDisplayedToStudents);
+
+        // This if statement should only be processed if only one instructor is displayed to students and you are
+        // making an instructor that was previously displayed to students invisible
+        if (numOfInstrDisplayed < 1 && !isDisplayedToStudents && instructorToEdit.isDisplayedToStudents) {
```

Comment:
In lieu of the long comment, you can introduce an explaining variable to make the code self-explanatory.
```java
boolean isSoleVisibleInstructor = numOfInstrDisplayed < 1 
boolean isMakingInstructorInvisible = !isDisplayedToStudents && instructorToEdit.isDisplayedToStudents;
if(isSoleVisibleInstructor && isMakingInstructorInvisible) {
    ...
```
By the way, why use `< 1` instead of `== 1` ?

Reply:
Also, you can check `isMakingInstructorInvisible` first and avoid calling the more expensive `getNumberOfInstructorsDisplayedToStudents(...)` if the result is false.